proto/iotics/api/meta.proto

```
@@ -0,0 +1,125 @@
+
+  // SPARQL request payload
+  Payload payload = 3;
+}
+
+// Part of a result for a SPARQL query request. Multiple chunks form a complete result. Related chunks can be identified
+// by a combination of:
+// - The host ID (unset for local results)
+// - Client reference (in headers, set by caller)
+// - Chunk sequence number
+// Note that chunks are guaranteed to arrive in-order (for a specific result from a host) but could be duplicated. It is
+// up to the client to ignore duplicates (based on sequence number).
+message SparqlQueryResultChunk {
+
+  // Payload of the query result chunk
+  message Payload {
+
+    // Result host identifier. Indicates from which host this result chunk came from. For a local result, this field
+    // will be unset.
+    HostID remoteHostId = 1;
+
+    // Position of a chunk in result from a given host (and request). The first chunk has a sequence number of 0.
+    uint64 seqNum = 2;
+
+    // Indicates whether this is the last chunk from a given host, for a specific request. Results for different
+    // requests can be identified by setting a unique clientRef in the request headers.
+    bool last = 3;
+
+    // Result error status (only applicable to local results, i.e. when remoteHostId is unset). If set, this will
+    // indicate a problem with running the query (e.g. invalid syntax) as opposed to a more general issue (in which case
+    // the standard gRPC error mechanism will be used and the stream terminated).
+    google.rpc.Status status = 4;
+
+    // The actual result content type (as raw mime type). For the hinted type (in a request), typically the returned
+    // type can be:
+    // - SPARQL_JSON - application/sparql-results+json
+    // - SPARQL_XML - application/sparql-results+xml
+    // - SPARQL_CSV - text/csv
+    // - RDF_TURTLE - text/turtle
+    // - RDF_XML - application/rdf+xml
+    // - RDF_NTRIPLES - application/n-triples
+    // - RDF_NQUADS - application/n-quads
+    string actualType = 5;
```

this may also need updated as per above if choosing to fail fast and not swap out types under the user. - contentType

---

@conbon - I'm well aware of that :+1:   - this would just be the enum

---

nice one.

---

e98aee517d91177a874f70b8cdec0821d66a7c75